Graphics/Slicer/Math/PGA.hs

```diff
@@ -74,50 +67,126 @@ plinesIntersectIn pl1 pl2
-  | onSegment l1 intersection && onSegment l2 intersection = IntersectsAt intersection
-  | scalarPart (rawPLine (eToPLine2 l1) ⎣ rawPLine (eToPLine2 l2)) ==  1 = Parallel
-  | scalarPart (rawPLine (eToPLine2 l1) ⎣ rawPLine (eToPLine2 l2)) == -1 = Parallel
-  | otherwise = NoIntersection
-  where
-    rawPLine (PLine2 a) = a
-    intersection = intersectionPoint l1 l2
-
 -- | Check if the second line's direction is on the 'left' side of the first line, assuming they intersect. If they don't intersect, return Nothing.
-lineIsLeft :: LineSeg -> LineSeg -> Maybe Bool
-lineIsLeft line1 line2
+pLineIsLeft :: PLine2 -> PLine2 -> Maybe Bool
+pLineIsLeft line1 line2
   | dualAngle dnpl1 dnpl2 == 0 = Nothing
   | otherwise                  = Just $ dualAngle dnpl1 dnpl2 > 0
   where
-    npl1 = dualPLine2 $ normalizePLine2 $ eToPLine2 line1
-    npl2 = dualPLine2 $ normalizePLine2 $ eToPLine2 line2
+    npl1 = dualPLine2 $ normalizePLine2 line1
+    npl2 = dualPLine2 $ normalizePLine2 line2
     (PPoint2 dnpl1) = forcePPoint2Basis $ PPoint2 npl1
     (PPoint2 dnpl2) = forcePPoint2Basis $ PPoint2 npl2
 
--- | find... something? FIXME: this is a total hack. returns positive for left, negaative for right, and +45 degrees returns the same as -45 degrees.
+-- | Find out where two lines intersect, returning a projective point.
+intersectionOf :: PLine2 -> PLine2 -> PPoint2
+intersectionOf pl1 pl2 = canonicalizePPoint2 $ meet2PLine2 pl1 pl2
+
+-- | Find... something?
+--   FIXME: this is a total hack. returns positive for left, negaative for right, and +45 degrees returns the same as -45 degrees.
 dualAngle :: GVec -> GVec -> ℝ
 dualAngle ln1 ln2 = valOf 0 $ getVals [GEZero 1, GEZero 1, GEPlus 1, GEPlus 2] $ (\(GVec a) -> a) $ unlikeVecPair ln1 ln2
 
--- | find the angle of intersection of two PLines.
+-- | Find the angle of intersection of two PLines.
 angleBetween :: PLine2 -> PLine2 -> ℝ
 angleBetween pl1 pl2 =  scalarPart $ pv1 ⎣ pv2
   where
     (PLine2 pv1) = forcePLine2Basis (normalizePLine2 pl1)
     (PLine2 pv2) = forcePLine2Basis (normalizePLine2 pl2)
 
+-- | Translate a line a given amound along it's perpendicular bisector.
+translatePerp :: PLine2 -> ℝ -> PLine2
+translatePerp pl1 d = PLine2 $ addVecPair m $ rawPLine pl1
+  where
+    m = GVec [GVal (d*normOfPLine2 pl1) [GEZero 1]]
+    rawPLine (PLine2 a) = a
+
+-- | find the distance between a point and a line.
+distancePPointToPLine :: PPoint2 -> PLine2 -> ℝ
+distancePPointToPLine point line = normOfPLine2 $ join2PPoint2 point linePoint
+--error $ "result: " <> show (normOfPLine2 $ newLine) <> "\nLine: " <> show lvec <> "\nPoint: " <> show pvec <> "\nPerpLine: " <> show perpLine <> "\nLinePoint: " <> show linePoint <> "\nnewLine: " <> show newLine <> "\n"
+  where
+    (PLine2 lvec)  = normalizePLine2 line
+    (PPoint2 pvec) = canonicalizePPoint2 point
+    perpLine       = PLine2 $ lvec ⨅ pvec
```

Comment:
The `PLine2` type doesn't seem like it's doing much work for you if you are packing *and* unpacking it in the same where block.